server/src/main/java/org/opensearch/extensions/ExtensionDependency.java

```diff
@@ -0,0 +1,89 @@
+/*
+* Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.extensions;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import org.opensearch.Version;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+
+/**
+ * The extension dependency information
+ *
+ * @opensearch.internal
+ */
+public class ExtensionDependency implements Writeable {
```

Comment:
```suggestion
public class DependentExtension implements Writeable {
```

Reply:
Yeah, don't want to hold PR for the same. `DependentExtensions` provides a better understanding of this class in my opinion. But again, it's my view.

Reply:
How about let us keep `ExtensionDependency` first, after we make decision, I will make the change in next PR.